prime-dotnet-webapi/Services/EnrolleePaperSubmissionService.cs

```diff
@@ -264,5 +264,126 @@ public async Task FinalizeSubmissionAsync(int enrolleeId)
             _context.Set<T>().RemoveRange(oldItems);
             _context.Set<T>().AddRange(itemList);
         }
+
+        public async Task<bool> PotentialReturneeExistsAsync(DateTime dateOfBirth)
+        {
+            var confirmedLinks = await _context.EnrolleeLinkedEnrolment
+                .AsNoTracking()
+                .Where(cl => cl.Confirmed)
+                .Select(cl => cl.PaperEnrolleeId)
+                .ToListAsync();
+
+            return await _context.Enrollees
+                .AsNoTracking()
+                .AnyAsync(
+                    e => e.GPID.StartsWith(PaperGpidPrefix)
+                    && !confirmedLinks.Contains(e.Id)
+                    && e.DateOfBirth.Date == dateOfBirth.Date
+                );
+        }
+
+        public async Task<IEnumerable<Enrollee>> GetPotentialPaperEnrolleeReturneesAsync(DateTime dateOfBirth)
+        {
+            var confirmedLinks = await _context.EnrolleeLinkedEnrolment
+                .AsNoTracking()
+                .Where(cl => cl.Confirmed)
+                .Select(cl => cl.PaperEnrolleeId)
+                .ToListAsync();
+
+            return await _context.Enrollees
+                .AsNoTracking()
+                .Where(
+                    e => e.GPID.StartsWith(PaperGpidPrefix)
+                    && e.DateOfBirth.Date == dateOfBirth.Date
+                    && !confirmedLinks.Contains(e.Id)
+                )
+                .ToListAsync();
+        }
+
+        public async Task<bool> LinkEnrolmentToPaperEnrolmentAsync(int enrolmentId, int paperEnrolmentId, bool isConfirmed = false)
```

Comment:
Whenever I have a method that has multiple parameters in a row of the same type, I tend to like calling them with named parameters to avoid accidentally calling the method with the parameters in the wrong order.